shell/platform/android/io/flutter/view/AccessibilityViewEmbedder.java

```diff
@@ -0,0 +1,319 @@
+ * continues working but the accessibility information for the platform view will not be embedded.
+ *
+ * We use the term `flutterId` for virtual accessibility node IDs in the FlutterView tree, and the term `originId`
+ * for the virtual accessibility node IDs in the platform view's tree. Internally this class maintains a bidirectional
+ * mapping between `flutterId`s and the corresponding platform view and `originId`.
+ */
+class AccessibilityViewEmbedder {
+    private static final String TAG = "AccessibilityBridge";
+
+    private final ReflectionAccessors reflectionAccessors;
+
+    // The view to which the platform view is embedded, this is typically FlutterView.
+    private final View rootAccessibilityView;
+
+    // Maps a flutterId to the corresponding platform view and originId.
+    private final SparseArray<ViewAndId> flutterIdToOrigin;
+
+    // Maps a platform view and originId to a corresponding flutterID.
+    private final Map<ViewAndId, Integer> originToFlutterId;
+
+    // Maps the flutterId of an accessibility node to the screen bounds of
+    // the root semantic node for the embedded view.
+    // This is used to translate the coordinates of the accessibility node subtree to the main display's coordinate
+    // system.
+    private final SparseArray<Rect> flutterIdToDisplayBounds;
+
+    private int nextFlutterId;
+
+    AccessibilityViewEmbedder(View rootAccessibiiltyView, int firstVirtualNodeId) {
+        reflectionAccessors = new ReflectionAccessors();
+        flutterIdToOrigin = new SparseArray<>();
+        this.rootAccessibilityView = rootAccessibiiltyView;
+        nextFlutterId = firstVirtualNodeId;
+        flutterIdToDisplayBounds = new SparseArray<>();
+        originToFlutterId = new HashMap<>();
+    }
+
+    /**
+     * Returns the root accessibility node for an embedded platform view.
+     *
+     * @param flutterId the virtual accessibility ID for the node in flutter accessibility tree
+     * @param displayBounds the display bounds for the node in screen coordinates
+     */
+    public AccessibilityNodeInfo getRootNode(View embeddedView, int flutterId, Rect displayBounds) {
+        AccessibilityNodeInfo originNode = embeddedView.createAccessibilityNodeInfo();
+        Long originPackedId = reflectionAccessors.getSourceNodeId(originNode);
+        if (originPackedId == null) {
+            return null;
+        }
+        int originId = ReflectionAccessors.getVirtualNodeId(originPackedId);
+        flutterIdToOrigin.put(flutterId, new ViewAndId(embeddedView, originId));
+        flutterIdToDisplayBounds.put(flutterId, displayBounds);
+        originToFlutterId.put(new ViewAndId(embeddedView, originId), flutterId);
+        return convertToFlutterNode(originNode, flutterId, embeddedView);
+    }
+
+    /**
+     * Returns the accessibility node info for the node identified with `flutterId`.
+     */
+    public AccessibilityNodeInfo createAccessibilityNodeInfo(int flutterId) {
```

Comment:
The comment vs name here seems a bit confusing.  The comment mentions "the accessibility node info" but the name of the method refers to "create".  Can you either update the comment to reflect creation, or the name of the method to reflect retrieval?

Reply:
done